### PR TITLE
Add sphinx

### PR DESCRIPTION
**Pull request by ghost**
- Add docs role including sphinx
- Add genshi
- Add python role for use by any role using python pip/venvs

### CODE REVIEW

File: ansible/roles/omero-build/tasks/main.yml [hunk 1]
```diff
@@ -35,6 +35,13 @@
     - python-pip
     - python-virtualenv
 
+- name: system packages | pip install packages
+  pip:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - genshi
```
Comment (sbesson) [inline ansible/roles/omero-build/tasks/main.yml]: Why do we need this dependency in the OMERO-build role?

Comment (rleigh-dundee) [inline ansible/roles/omero-build/tasks/main.yml]: For reasons I'm not entirely sure about, this role is also installing all the bio-formats dependencies as well.

Comment (sbesson) [inline ansible/roles/omero-build/tasks/main.yml]: Understood. We might want to split omero-build from bf-build eventually unless there is a reason to keep them together /cc @manics @kennethgillen

Comment (rleigh-dundee) [inline ansible/roles/omero-build/tasks/main.yml]: At the moment the hosts are generally building both, so the need for splitting isn't that great, but certainly can do so if it makes sense.

**Comment by manics** [comment]
Looks generally OK. Rename `docs` roel to somethin more descriptive? e.g. `omero-docs`, `omero-docs-build`, `omero-sphinx`, etc depending on the overall plan.

**Comment by rleigh-dundee** [comment]
The docs is fairly generic, but really "what you need for sphinx", so could be renamed to e.g. `sphinx-build`?

**Comment by rleigh-dundee** [comment]
@manics Now renamed.

**Comment by joshmoore** [comment]
NB: elsewhere we've actively removed genshi from installs. cc: @sbesson

**Comment by rleigh-dundee** [comment]
@joshmoore While this is true for the build from git and the full-fat release zip, it's deliberately not the case for the -dfsg cleaned up source zip, which we still need to build and test.

**Comment by joshmoore** [comment]
Hmmm.... interesting, though since that means we can only cleanly test one of the two variants. Along with ome/ome-cmake-superbuild#8 it might be good to discuss what's-being-used+tested-where.

**Comment by rleigh-dundee** [comment]
@joshmoore The super build PR will install genshi and so provide a means of testing it without it having to be installed on the base system.  At the moment though the centos7 node is the exception--all the others have it installed.

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [INFRASTRUCTURE-merge#1](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/1/). See the [console output](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/1/consoleText) for more details.
Possible conflicts:
  - PR #8 kennethgillen 'Added role Texlive'
    - ansible/ci-deployment.yml

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [INFRASTRUCTURE-merge#2](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/2/). See the [console output](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/2/consoleText) for more details.
Possible conflicts:
  - PR #8 kennethgillen 'Added role Texlive'
    - ansible/ci-deployment.yml

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [INFRASTRUCTURE-merge#3](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/3/). See the [console output](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/3/consoleText) for more details.
Possible conflicts:
  - PR #8 kennethgillen 'Added role Texlive'
    - ansible/ci-deployment.yml

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [INFRASTRUCTURE-merge#4](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/4/). See the [console output](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/4/consoleText) for more details.
Possible conflicts:
  - PR #8 kennethgillen 'Added role Texlive'
    - ansible/ci-deployment.yml

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [INFRASTRUCTURE-merge#5](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/5/). See the [console output](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/5/consoleText) for more details.
Possible conflicts:
  - PR #8 kennethgillen 'Added role Texlive'
    - ansible/ci-deployment.yml

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [INFRASTRUCTURE-merge#6](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/6/). See the [console output](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/6/consoleText) for more details.
Possible conflicts:
  - PR #8 kennethgillen 'Added role Texlive'
    - ansible/ci-deployment.yml

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [INFRASTRUCTURE-merge#7](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/7/). See the [console output](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/7/consoleText) for more details.
Possible conflicts:
  - PR #8 kennethgillen 'Added role Texlive'
    - ansible/ci-deployment.yml

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [INFRASTRUCTURE-merge#8](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/8/). See the [console output](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/8/consoleText) for more details.
Possible conflicts:
  - PR #8 kennethgillen 'Added role Texlive'
    - ansible/ci-deployment.yml

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [INFRASTRUCTURE-merge#9](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/9/). See the [console output](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/9/consoleText) for more details.
Possible conflicts:
  - PR #8 kennethgillen 'Added role Texlive'
    - ansible/ci-deployment.yml

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [INFRASTRUCTURE-merge#10](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/10/). See the [console output](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/10/consoleText) for more details.
Possible conflicts:
  - PR #8 kennethgillen 'Added role Texlive'
    - ansible/ci-deployment.yml

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [INFRASTRUCTURE-merge#11](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/11/). See the [console output](https://ci.openmicroscopy.org/job/INFRASTRUCTURE-merge/11/consoleText) for more details.
Possible conflicts:
  - PR #8 kennethgillen 'Added role Texlive'
    - ansible/ci-deployment.yml

**Comment by joshmoore** [comment]
--exclude to reduce noise.

**Comment by sbesson** [comment]
@rleigh-dundee: you should be able to fix the conflict on this PR now.

**Comment by mtbc** [comment]
Still issues here. Once conflicts are fixed may now need an `include` label? Though maybe best to just open a new PR.

**Comment by sbesson** [comment]
I won't have time to get back to this anytime this week. Closing for now. The refactoring of the Sphinx dependenceis were also discussed in https://github.com/openmicroscopy/infrastructure/pull/129. Maybe the good moment to test this is when we work on removing the PDF dependency.

**Pull request closed without merging**